File: rr2/code/rr2/generate_pdfs/shopping_list.rb
#---
# Excerpted from "Rails Recipes",
# published by The Pragmatic Bookshelf.
# Copyrights apply to this code. It may not be used to create training material, 
# courses, books, articles, and the like. Contact us if you are in doubt.
# We make no guarantees that this code is fit for any purpose. 
# Visit http://www.pragmaticprogrammer.com/titles/rr2 for more book information.
#---
shopping_list = [
  ["Carton of Goat Milk", 1],
  ["Head of Garlic", 2],
  ["Chocolate Bar", 9]
]
Prawn::Document.generate("shopping_list.pdf") do
  table([[ "Item", "Quantity" ], *shopping_list]) do  |t|
    t.header = true
    t.row_colors = [ "aaaaff", "aaffaa", "ffaaaa" ]
    t.row(0).style :background_color => '448844', :text_color => 'ffffff'
    t.columns(1).align = :right
  end
end

File: rcctr/code/ruby/groceries.rails2/public/javascripts/shoppingList.js
/***
 * Excerpted from "Continuous Testing",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material, 
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose. 
 * Visit http://www.pragmaticprogrammer.com/titles/rcctr for more book information.
***/
modit('app.view', function() {
  function shoppingList(div) {
    var list = $('<li>');
    var items = $.get('menus/6.json');
    _.each(items, function(item) {
      list.append($('<li>').addClass('shopping-list-item'));
    });
    div.append(list);
  }
  this.exports(shoppingList);
});

modit('app.view', function() {
  function shoppingList(div) {
    var list = $('<ul>');
    var menu = $.getJSON('menus/6.json');
    _.each(menu.ingredients, function(item) {
      list
        .append($('<li>')
          .append($('<div>')
            .addClass('shopping-list-item')
            .text(item.quantity + " " + item.unit + " " + item.name)
          )
        );
    });
    div.append(list);
  }
  this.exports(shoppingList);
});
